2016-05/may-12.md

```diff
@@ -0,0 +1,107 @@
+* Sentry (and others!) can look up that number in a map.
+* A new intern will be working on this!
+
+### New Intern!
+
+* Keyan will be working on the React team as an intern.
+* [Ben](https://twitter.com/soprano) will be mentoring him.
+* The plan is for him to work on the error code system described above.
+* Another thing he might work on is a “yellow box” a la React Native.
+
+#### Yellow Box
+
+* Where should it go? fbjs?
+* Could be a more effective vector for communication between the React core team and users of React than console warnings.
+* To be practical, we would need to have warning levels and some way of suppressing them.
+* This is just an idea, the details would need to be fleshed out.
+
+### `shouldComponentUpdate()` Breaks Context
+
+* A [longstanding issue](https://github.com/facebook/react/issues/2517) with a very long thread.
+* [Sebastian](https://twitter.com/sebmarkbage) has been thinking about this (just this morning!)
+* With the [incremental reconciler](https://github.com/facebook/react/issues/6170), we think we’ll solve this.
+
+#### Can We Fix It Now?
+
+* It will make things a lot more inefficient.
+* Only some libraries use context for dynamically updating values, but enabling deep subscriptions by default will hurt performance for everyone.
+
+#### Hacky Workaround
+
+* For now, React Router 3.0 adds a hacky [workaround](https://github.com/reactjs/react-router/blob/5f3387a91d427aec1368e68cd9605797aa076bb7/modules/ContextUtils.js).
+* [@taion](https://github.com/taion) plans to pull it out in a generic module for other affected libraries.
+* Nobody is really happy about it, but hopefully it should be easy to pull it out later.
+* It [uses mixins](https://github.com/reactjs/react-router/issues/3439#issuecomment-217887475) so it’s easier to pull it out, and it doesn’t trash React DevTools view.
+* Hopefully eventually React fixes this, and React Router can <s>unpublish</s> stop using that package.
+
+### Incremental Progress vs Long-Term Vision
+
+* Long-term ideas and features often block incremental and external contributions.
+* We might want to spend more time just supporting features the community is asking for, even if we don't use them.
+* One concern is that the proposed solution often will leave us in a worse place than where we were before.
+* This is often not obvious in the beginning, and unfortunately a lot of work may get done before we realize this.
+
+#### Focus on One Thing
+
+* Maybe as a team we can focus on one area at a time.
+* For example, let’s just do one thing, do it well, and ship it.
+* Better than letting ourselves get overwhelmed.
+
+#### Avoid Wasted Effort
+
+* We can focus on external needs, but we also need to make sure that external folks are willing to support our needs internally.
+* This requires that we better communicate our needs and priorities.
+* There’s a long unwritten list of constraints, and we often reactively provide that list.
```

Comment:
reactively ... pun pun :-)

sometimes I can put together a picture of what the team is focusing on, what's in flight, etc., by traversing PRs and Issues through their related links, but it is quite time consuming (Not that I have a solution ;-) )